assessment_software/src/app/add-course/add-course.component.html

```diff
@@ -0,0 +1,27 @@
+<div class="form-container">
+    <form [formGroup]="addCourseForm" (submit)="submitCourse()">
```

Comment:
here you are doing (submit)="submitCourse()", so you have to make the button type=submit so that submitCourse() is called. 

that is one way to do it. the other way would be to use the button callback and call submitCourse from there. 

```
<app-button variant="Standard" color="gray" onClick="handleCancel">Cancel</app-button>
 <app-button variant="Filled" color="white" onClick="handleSubmit" >Confirm</app-button>
```

then, from handleSubmit function, call submitCourse();